clustered/server/src/main/java/org/ehcache/clustered/server/state/MessageTracker.java

```diff
@@ -66,7 +65,7 @@ public void track(long msgId) {
    * @return true if the given msgId is already tracked otherwise false.
    */
   public boolean seen(long msgId) {
-    boolean seen = (msgId <= highestContiguousMsgId) || nonContiguousMsgIds.contains(msgId);
+    boolean seen = nonContiguousMsgIds.contains(msgId) || msgId <= highestContiguousMsgId;
```

Comment:
This change makes it very inefficient. Since we can't really have a `track` and `seen` for the same `msgId` concurrently, maybe we should have a fast path then the full path, like:
```java
boolean seen = true;
if (msgId > highestContiguousMsgId) {
  seen = nonContiguousMsgIds.contains(msgId) || msgId <= highestContiguousMsgId;
}
```